index.js

```diff
@@ -2,6 +2,16 @@
 var _ = require('lodash');
 
 module.exports = function (inputObj, opts) {
+  // Throw error if `inputObj` does not exist or is not an array/object.
+  if ( !inputObj || typeof inputObj !== 'object'  ) {
+    throw new Error( 'Whoops! Please ensure that `barHorizontal()` is invoked with either an array or object.' );
+  }
+
+  // Throw error if `inputObj` is an empty array or key-less object.
+  if ( ( typeof inputObj.length !== 'undefined' && !inputObj.length ) || ( typeof inputObj.length === 'undefined' && !Object.keys( inputObj ).length ) ) {
```

Comment:
Is this `typeof` check required? Now that we know `inputObj` is definitely an object, wouldn't it be enough to ensure that the length isn't 0?

Reply:
I have mixed feelings about this block as well.

I was thinking that at this point, we don't actually know if the `inputObj` is an object or an array, and object literals don't expose a `length` property. Maybe the whole condition is not required (eg. DON'T error if the `inputObj` is empty/keyless)?

Reply:
What do you think about:

```js
typeof inputObj === 'object' && inputObj != null && Object.keys(inputObj).length > 0
```

(Arrays are just json objects) I think this should be [enough](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#Examples)

And the error message could be `Invoke bar-horizontal with a non-empty array or object`. This single check would replace both your `if` statements.

P.S lodash's `isEmpty` also does all this checking. But it is overkill in this case.

Reply:
Awesome, thanks for the direction. Will update.